armada/handlers/armada.py

```diff
@@ -131,13 +131,17 @@ def sync(self):
                 installed_chart, installed_values = self.find_release_chart(
                     known_releases, release_prefix(prefix, chart.release_name))
 
-                if not self.disable_update_pre and getattr(chart, 'upgrade',
-                                                           False):
-                    pre_actions = getattr(chart.upgrade, 'pre', {})
-
-                if not self.disable_update_post and getattr(chart, 'upgrade',
-                                                            False):
-                    post_actions = getattr(chart.upgrade, 'post', {})
+                LOG.info("Checking Pre/Post Actions")
+                upgrade = entry.get('chart').get('upgrade', False)
```

Comment:
```python

upgrade = entry.get('chart').get(...)
```
The code above can result in errors if 'chart' attribute is not found or isn't a dict. I'd change it to:

```python

upgrade = entry.get('chart', {}).get(...)
```